manager/container.go

```diff
@@ -78,6 +83,17 @@ type containerData struct {
 	collectorManager collector.CollectorManager
 }
 
+// jitter returns a time.Duration between duration and duration + maxFactor * duration,
+// to allow clients to avoid converging on periodic behavior.  If maxFactor is 0.0, a
+// suggested default value will be chosen.
+func jitter(duration time.Duration, maxFactor float64) time.Duration {
```

Comment:
Making it variable might invalidate assumptions made over cAdvisor APIs.
For example, heapster's internal logic might break if the housekeeping
jitter is arbitrarily large.

On Fri, Jan 15, 2016 at 2:54 PM, Timothy St. Clair <notifications@github.com
> wrote:

> In manager/container.go
> <https://github.com/google/cadvisor/pull/1057#discussion_r49915178>:
>
> > @@ -78,6 +83,17 @@ type containerData struct {
> >  	collectorManager collector.CollectorManager
> >  }
> >
> > +// jitter returns a time.Duration between duration and duration + maxFactor * duration,
> > +// to allow clients to avoid converging on periodic behavior.  If maxFactor is 0.0, a
> > +// suggested default value will be chosen.
> > +func jitter(duration time.Duration, maxFactor float64) time.Duration {
>
> I think we should keep the function parameter at minimum, and if we want
> to remove the constant and simply default the argument that's fine. But at
> this point we're splitting hairs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1057/files#r49915178>.
>